package/pom.xml

```diff
@@ -63,19 +63,35 @@
   <build>
     <plugins>
       <plugin>
-        <artifactId>maven-assembly-plugin</artifactId>
-        <configuration>
-          <finalName>
-            ${jar.assembly.name.prefix}-spark${sparkbundle.version}_${scala.binary.version}-${project.version}
-          </finalName>
-        </configuration>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
         <executions>
           <execution>
-            <id>make-assembly</id>
             <phase>package</phase>
             <goals>
-              <goal>single</goal>
+              <goal>shade</goal>
             </goals>
+            <configuration>
+              <relocations>
+                <relocation>
+                  <pattern>com.google.protobuf</pattern>
+                  <shadedPattern>io-glutenproject.shaded.com.google.protobuf</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.apache.arrow</pattern>
+                  <shadedPattern>io-glutenproject.shaded.org.apache.arrow</shadedPattern>
+                  <!--arrow's C wrapper refers to the original class path, so we should not relocate here-->
+                  <excludes>
+                    <exclude>org.apache.arrow.c.*</exclude>
+                    <exclude>org.apache.arrow.c.jni.*</exclude>
+                  </excludes>
+                </relocation>
```

Comment:
:+1: 

So Spark doesn't refer to `arrow-c-data` jar, right?

Reply:
I don't think it is doing it at least right now